src/lib/cosmos/components/index.tsx

```diff
@@ -0,0 +1,222 @@
+          setBalances(balances)
+        }
+      })
+
+      return () => {
+        stale = true
+        setBalances(undefined)
+      }
+    }
+  }, [provider, accounts])
+
+  return balances
+}
+
+function Accounts({
+  useAnyNetwork,
+  hooks: { useAccounts, useProvider, useENSNames },
+}: {
+  useAnyNetwork: boolean
+  hooks: Web3ReactHooks
+}) {
+  const provider = useProvider(useAnyNetwork ? 'any' : undefined)
+  const accounts = useAccounts()
+  const ENSNames = useENSNames(provider)
+
+  const balances = useBalances(provider, accounts)
+
+  return (
+    <div>
+      Accounts:
+      {accounts === undefined
+        ? ' -'
+        : accounts.length === 0
+        ? ' None'
+        : accounts?.map((account, i) => (
+            <ul key={account} style={{ margin: 0, overflow: 'hidden', textOverflow: 'ellipsis' }}>
+              <b>{ENSNames?.[i] ?? account}</b>
+              {balances?.[i] ? ` (Ξ${formatEther(balances[i])})` : null}
+            </ul>
+          ))}
+    </div>
+  )
+}
+
+function Connect({
+  connector,
+  hooks: { useChainId, useIsActivating, useError, useIsActive },
+}: {
+  connector: Connector
+  hooks: Web3ReactHooks
+}) {
+  const chainId = useChainId()
+  const isActivating = useIsActivating()
+  const error = useError()
+
+  const active = useIsActive()
+
+  const [activateArgs] = useState<any[]>([])
+
+  if (error) {
+    return <button onClick={() => connector.activate()}>Try Again?</button>
+  } else if (active) {
+    return (
+      <>
+        {connector instanceof Network ? (
+          <label>
+            Network:
+            <select value={`${chainId}`} onChange={(event) => connector.activate(Number(event.target.value))}>
+              <option value={SupportedChainId.MAINNET}>Mainnet</option>
+              <option value={SupportedChainId.ROPSTEN}>Ropsten</option>
+              <option value={SupportedChainId.RINKEBY}>Rinkeby</option>
+              <option value={SupportedChainId.GOERLI}>Görli</option>
+              <option value={SupportedChainId.KOVAN}>Kovan</option>
+              <option value={SupportedChainId.OPTIMISM}>Optimism</option>
+              <option value={SupportedChainId.ARBITRUM_ONE}>Arbitrum</option>
+            </select>
+          </label>
+        ) : null}
+        <button
+          onClick={() => {
+            if (connector.deactivate) {
+              connector.deactivate()
+            }
+          }}
+          disabled={!connector.deactivate}
+        >
+          {connector.deactivate ? 'Disconnect' : 'Connected'}
+        </button>
+      </>
+    )
+  } else {
+    return (
+      <button
+        onClick={() => {
+          if (!isActivating) {
+            connector.activate(...activateArgs)
+          }
+        }}
+        disabled={isActivating}
+      >
+        {isActivating ? 'Connecting...' : 'Activate'}
+      </button>
+    )
+  }
+}
+
+const Wrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+`
+const ConnectorWrapper = styled.div`
```

Comment:
I'd like this to be styled more like the cosmos interface, so it's more obviously part of the configurator and not a part of the fixture being viewed. WDYT?

Reply:
sure, seems reasonable